### PR TITLE
Added 'mode' prop to SignInButton and SignUpButton component docs

### PR DESCRIPTION
**Pull request by royanger**
Added info on using the components as modals.

### CODE REVIEW

**Comment by royanger** [comment]
I'm not sure if there is nuance for frameworks other than Next.js.

File: docs/components/unstyled/sign-in-button.mdx [hunk 1]
```diff
@@ -156,6 +156,13 @@ In some cases you will want to use your own button, or button text. You can do t
         <code>string</code>,
         <>Full URL or path to navigate after successful sign up.</>,
       ],
+    },
+    {
+      cells: [
+        <code>mode</code>,
+        <code>"redirect" | "modal"</code>,
+        <><code>redirect</code> is the default behaviour and will redirect to sign-in route. <code>modal</code> will open a modal one current route.</>
```
Comment (alexisintech) [inline docs/components/unstyled/sign-in-button.mdx:164]:
```suggestion
        <code>'redirect' | 'modal'</code>,
        <>Determines what happens when a user clicks on the <code><SignInButton /></code>. Setting this to <code>'redirect'</code> will redirect the user to the sign-in route. Setting this to <code>'modal'</code> will open a modal on the current route.<br />Defaults to <code>'redirect'</code>.</>
```

File: docs/components/unstyled/sign-up-button.mdx [hunk 2]
```diff
@@ -164,6 +164,13 @@ In some cases you will want to use your own button, or button text. You can do t
         <code>string</code>,
         <>Full URL or path to navigate after successful sign up.</>,
       ],
+    },
+    {
+      cells: [
+        <code>mode</code>,
+        <code>"redirect" | "modal"</code>,
+        <><code>redirect</code> is the default behaviour and will redirect to sign-up route. <code>modal</code> will open a modal one current route.</>
```
Comment (alexisintech) [inline docs/components/unstyled/sign-up-button.mdx:172]:
```suggestion
        <code>'redirect' | 'modal'</code>,
        <>Determines what happens when a user clicks on the <code><SignUpButton /></code>. Setting this to <code>'redirect'</code> will redirect the user to the sign-in route. Setting this to <code>'modal'</code> will open a modal on the current route.<br />Defaults to <code>'redirect'</code>.</>
```

**Review by alexisintech — CHANGES_REQUESTED** [review]

**Review by BRKalow — APPROVED** [review]

File: docs/components/unstyled/sign-in-button.mdx [hunk 3]
```diff
@@ -156,6 +156,13 @@ In some cases you will want to use your own button, or button text. You can do t
         <code>string</code>,
         <>Full URL or path to navigate after successful sign up.</>,
       ],
+    },
+    {
+      cells: [
+        <code>mode</code>,
+        <code>'redirect' | 'modal'</code>,
+        <>Determines what happens when a user clicks on the <code><SignInButton /></code>. Setting this to <code>'redirect'</code> will redirect the user to the sign-in route. Setting this to <code>'modal'</code> will open a modal on the current route.<br />Defaults to <code>'redirect'</code>.</>
```
Comment (BRKalow) [inline docs/components/unstyled/sign-in-button.mdx:164]:
```suggestion
        <>Determines what happens when a user clicks on the <code>{'<SignInButton />'}</code>. Setting this to <code>'redirect'</code> will redirect the user to the sign-in route. Setting this to <code>'modal'</code> will open a modal on the current route.<br />Defaults to <code>'redirect'</code>.</>
```

File: docs/components/unstyled/sign-up-button.mdx [hunk 4]
```diff
@@ -164,6 +164,13 @@ In some cases you will want to use your own button, or button text. You can do t
         <code>string</code>,
         <>Full URL or path to navigate after successful sign up.</>,
       ],
+    },
+    {
+      cells: [
+        <code>mode</code>,
+        <code>'redirect' | 'modal'</code>,
+        <>Determines what happens when a user clicks on the <code><SignUpButton /></code>. Setting this to <code>'redirect'</code> will redirect the user to the sign-in route. Setting this to <code>'modal'</code> will open a modal on the current route.<br />Defaults to <code>'redirect'</code>.</>
```
Comment (BRKalow) [inline docs/components/unstyled/sign-up-button.mdx:172]:
```suggestion
        <>Determines what happens when a user clicks on the <code>{'<SignUpButton />'}</code>. Setting this to <code>'redirect'</code> will redirect the user to the sign-in route. Setting this to <code>'modal'</code> will open a modal on the current route.<br />Defaults to <code>'redirect'</code>.</>
```

File: docs/components/unstyled/sign-in-button.mdx (hunk 3 above)
Comment (alexisintech) [reply docs/components/unstyled/sign-in-button.mdx]: And this is why it needs to be converted to a markdown table 🙈

**Pull request merged**